### PR TITLE
Add unbounded_knapsack.py

### PR DESCRIPTION
**Pull request by kenzonobre**
### Describe your change:

Add unbounded knapsack algorithm, which is an extension of the 0-1 knapsack problem.

0-1 Knapsack Problem :
https://en.wikipedia.org/wiki/Knapsack_problem
https://www.geeksforgeeks.org/0-1-knapsack-problem-dp-10/

Unbounded knapsack : 
https://www.geeksforgeeks.org/unbounded-knapsack-repetition-items-allowed/

* [X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [X] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [X] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by kenzonobre** [comment]
@cclauss could you please review my code? I could work on it more if necessary, but I think it looks good

**Review by CaedenPH — APPROVED** [review]
Looks good to me, but I'm sure Cclauss will find some problems ;)

**Comment by kenzonobre** [comment]
@cclauss could you please review my code? I could work on it more if necessary, but I think it looks good

File: knapsack/unbounded_knapsack.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    >>> items_weights = [200, 400]
+    >>> knapsack(c, items_weights, items_values)
+    0
+
+    The result is 0 because no items can be carried.
+    """
+
+    # n is the number of items
+    n = len(items_weights)
+
```
Comment (cclauss) [inline knapsack/unbounded_knapsack.py:57]:
```suggestion
```

File: knapsack/unbounded_knapsack.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+def knapsack(capacity: int, items_weights: list[int], items_values: list[int]) -> int:
+    """
+    Returns the maximum value of items that can be put in a knapsack of a capacity c.
+    Each type of item has a specific weight and value.
+    It is possible to carry unlimited number of instances of an item.
+
+    >>> c = 70
+    >>> items_values = [40, 900, 120]
+    >>> items_weights = [10, 20, 30]
+    >>> knapsack(c, items_weights, items_values)
```
Comment (cclauss) [inline knapsack/unbounded_knapsack.py:30]:
```suggestion
    >>> knapsack(capacity=70, weights=(40, 900, 120), values=(10, 20, 30))
```
Please repeat for the other tests.

Please add tests for:
`capacity = 0`
`capacity - -5`
`len(values)` != `len(weights)`
`len(values) == 5` and `len(weights) == 0`
etc.

**Review by cclauss — CHANGES_REQUESTED** [review]
* [ ] How is this different than all our other knapsack algorithms?
* [ ] Is it faster?  Can we prove that with a benchmark?
* [ ] How is this algorithm unbound if the first parameter is `capacity`?
* [ ] Please add a test that demonstrates how would this unbound algorithm would deal with `weights` and `values` that were never-ending `cycles`?  https://docs.python.org/3/library/itertools.html#itertools.cycle

**Pull request closed without merging**